sqlhooks.go

```diff
@@ -9,12 +9,27 @@ import (
 // Hook is the hook callback signature
 type Hook func(ctx context.Context, query string, args ...interface{}) (context.Context, error)
 
+// ErrorHandler is the error handling callback signature
+type ErrorHandler func(ctx context.Context, err error, query string, args ...interface{}) error
+
 // Hooks instances may be passed to Wrap() to define an instrumented driver
 type Hooks interface {
 	Before(ctx context.Context, query string, args ...interface{}) (context.Context, error)
 	After(ctx context.Context, query string, args ...interface{}) (context.Context, error)
 }
 
+// ErrHook instances will be called if any error happens
+type ErrHook interface {
+	OnError(ctx context.Context, err error, query string, args ...interface{}) error
+}
+
+func handlerErr(ctx context.Context, errHook ErrHook, err error, query string, args ...interface{}) error {
+	if e := errHook.OnError(ctx, err, query, args...); e != nil {
+		return e
+	}
+	return err
+}
```

Comment:
you could simplify the below code if you took a `Hook` instead of an `ErrHook`. Then if the `Hook` implements the error hook interface you use it, otherwise you just return the err.